docs/articles/documentation/how-it-works/README.md

```diff
@@ -3,151 +3,40 @@ layout: docs
+## Page Proxying: Bridging the Server-Client Gap
 
-With this hybrid approach, TestCafe tests combine advantages of the browser scripts (built-in waiting, mobile testing) and standalone server-side code (access to the app's server and testing environment, better control over test execution).
+You may notice that when you run TestCafe, the URL in the browser’s address bar does not match that of your website. This happens because TestCafe runs an under-the-hood reverse proxy.
 
-### Node.js API
+The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and adds automation scripts to the requested resources. It then modifies the URLs contained within the resource so that they point to the proxy. That way, neither the client-side code nor the resources it communicates with can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe mocks the browser API.
 
-Test code uses compact API that consists of just a few dozen methods that eventually transform to user actions on the page. Since this server-side code runs in Node.js, tests should be written in JavaScript or a language TestCafe can compile to JavaScript (like TypeScript or CoffeeScript).
+The proxying mechanism ensures that the page appears to be hosted at the original URL even to the test code. This is why you can use your website’s actual URL in tests, and pay no mind to the browser’s address bar.
 
-The concise API makes tests easy to write, read, and maintain. Support for popular web languages allows most users to use their existing skills and minimizes the learning curve for new users.
+## Browser Sandboxing
 
-### Page proxying
+Every TestCafe test begins with an empty slate of a browser. At the end of a run, TestCafe deletes all browser cookies, empties the storage, and reloads the page, thereby preventing undesirable interference with subsequent tests. You don’t need to write any boilerplate code to reset the app state and reverse the changes your tests make.
 
-To provide automation scripts (client-side code) to the browser, TestCafe runs a reverse proxy under the hood. It proxies the tested page on the server and injects the scripts into its code.
+Tests that run in parallel operate in independent sandboxed environments. This helps prevent server-side collisions.
 
-Script injection enables TestCafe key features like mobile testing, automatic waiting, or user roles for smart authentication out of the box.
+## Client-Side Scripts
 
-### Isolated tests
+To perform common testing tasks, TestCafe translates server-side calls to its API into client-side code. However, some scenarios require the execution of client-side code that TestCafe cannot automatically generate. That’s why the framework provides several ways to execute user-defined JavaScript.
```

Comment:
Ray dislikes 'provides' (as far as I know);
I think 'several' is not comprehensive. I'd replace it:

> That’s why the framework has more than one way to execute user-defined JavaScript.

Reply:
what about 'offers multiple ways'?